osf/management/commands/check_crossref_dois.py

```diff
@@ -67,7 +67,10 @@ def check_crossref_dois(dry_run=True):
         pending_dois = []
         for preprint in preprint_batch:
             prefix = preprint.provider.doi_prefix
-            pending_dois.append(f'doi:{settings.DOI_FORMAT.format(prefix=prefix, guid=preprint._id)}')
+            guid = preprint._id
+            if VersionedGuidMixin.GUID_VERSION_DELIMITER in preprint._id:
+                guid = preprint._id.split(VersionedGuidMixin.GUID_VERSION_DELIMITER)[0]
+            pending_dois.append(f'doi:{settings.DOI_FORMAT.format(prefix=prefix, guid=guid)}')
```

Comment:
Instead of splitting and grab the base GUID, you should fix the tests.

However, you can't fix the tests until you have the updated factory for Bohdan's PR: https://github.com/CenterForOpenScience/osf.io/pull/10840.

Can you comment below with the failed tests related this part of the code so you and collaborate with Bohdan to fix the DOI.